File: src/main/java/com/github/zxh/classpy/classfile/constant/ConstantNameAndTypeInfo.java
package com.github.zxh.classpy.classfile.constant;

/*
CONSTANT_NameAndType_info {
    u1 tag;
    u2 name_index;
    u2 descriptor_index;
}
*/
public class ConstantNameAndTypeInfo extends ConstantInfo {

    {
        u2("name_index");
        u2("descriptor_index");
    }

    public int getNameIndex() {
        return super.getInt("name_index");
    }

    @Override
    protected String loadDesc(ConstantPool cp) {
        String name = cp.getUtf8String(super.getInt("name_index"));
        String type = cp.getUtf8String(super.getInt("descriptor_index"));
        return name + "&" + type;
    }
    
}

File: src/main/kotlin/com/github/zxh/classpy/classfile/constant/ConstantNameAndTypeInfo.kt
package com.github.zxh.classpy.classfile.constant;

/*
CONSTANT_NameAndType_info {
    u1 tag;
    u2 name_index;
    u2 descriptor_index;
}
*/
class ConstantNameAndTypeInfo : ConstantInfo() {

    init {
        u2("name_index");
        u2("descriptor_index");
    }

    fun getNameIndex(): Int {
        return super.getInt("name_index");
    }

    override fun loadDesc(cp: ConstantPool): String {
        val name = cp.getUtf8String(super.getInt("name_index"));
        val type = cp.getUtf8String(super.getInt("descriptor_index"));
        return name + "&" + type;
    }
    
}
